lib/core.js

```diff
@@ -547,20 +549,22 @@ type IteratorResult<Yield,Return> =
   | { done: true, value?: Return }
   | { done: false, value: Yield };
 
-interface $Iterator<+Yield,+Return,-Next> {
-    @@iterator(): $Iterator<Yield,Return,Next>;
-    next(value?: Next): IteratorResult<Yield,Return>;
-}
-type Iterator<+T> = $Iterator<T,void,void>;
-
 interface $Iterable<+Yield,+Return,-Next> {
     @@iterator(): $Iterator<Yield,Return,Next>;
 }
 type Iterable<+T> = $Iterable<T,void,void>;
 
-interface Generator<+Yield,+Return,-Next> {
+interface $Iterator<+Yield,+Return,-Next> extends $Iterable<Yield,Return,Next> {
     @@iterator(): $Iterator<Yield,Return,Next>;
     next(value?: Next): IteratorResult<Yield,Return>;
+    +return?: <R>(value: R) => IteratorResult<Yield,R|Return>;
```

Comment:
A `return` definition would be great. :)

I'd think it should just be `() => IteratorResult<Yield, Return>` (same for `$AsyncIterator` below) as long as that doesn't break passing a generator where an iterator is expected.

If you're hand-writing a `return` method to handle abrupt completions, you normally wouldn't accept any value because none are passed on `IteratorClose`. See http://www.ecma-international.org/ecma-262/7.0/#sec-iteratorclose